### ISSUE TITLE
Support for "tag configure" in Entry/Textbox

### ISSUE THREAD

**Issue by samclane** [TomSchimansky/CustomTkinter#1504]
Writing a small code viewer, I used to use the functions:

```python
text_entry.tag_configure("Keyword", foreground=monokai.blue)
text_entry.tag_configure("String", foreground=monokai.green)
text_entry.tag_configure("Comment", foreground=monokai.comments)
```

However CtkTextbox/CtkEntry doesn't have these functions. Is there a way for me to dynamically color segments of text like this, or do I need to wait for it to be added to Ctk?

**Comment by Kurzuxus** [comment]
I have been working on a personal project and I did encounter an issue similar to this. I managed to do it this way and hopefully it will work for you:

```python
text_box=ctk.CTkTextbox(nest_fr2,width=400,height=300,wrap='word') #Create a textbox widget
text_box.pack(pady=10)

colors=['red','blue','green']

def change_color(color):
    selected text=text_box.tag_ranges('sel') #We get the range of highlighted text in the textbox
    current_color=option_menu.get() #The current color selected in the option menu
    if selected_text: # add this condition so when the file first starts it doesn't give an error that a text is not highlighted 
      tag_name='color_'+current_color
      text_box.tag_config(tag_name, foreground=current_color)
      for tag in text_box.tag_names(selected_text[0]):
        text_box.tag_remove(tag,selected_text[0],selected_text[1] #We remove a new tag in case there is already an existing tag
      text_box.tag_add(tag_name, selected_text[0],selected_text[1]) #We add the tag to the selected text with the new foreground color

option_menu=ctk.CTkOptionMenu(root, values=[x for x in colors], command=lambda color: change_color(color))
option_menu.pack(pady=10)
``` 
I'm not sure if this is particularly what you want but this is how I used the tag methods when working with the textbox widget. Hope it helps!

**Comment by Zingzy** [comment]
> I have been working on a personal project and I did encounter an issue similar to this. I managed to do it this way and hopefully it will work for you:
> 
> ```python
> text_box=ctk.CTkTextbox(nest_fr2,width=400,height=300,wrap='word') #Create a textbox widget
> text_box.pack(pady=10)
> 
> colors=['red','blue','green']
> 
> def change_color(color):
>     selected text=text_box.tag_ranges('sel') #We get the range of highlighted text in the textbox
>     current_color=option_menu.get() #The current color selected in the option menu
>     if selected_text: # add this condition so when the file first starts it doesn't give an error that a text is not highlighted 
>       tag_name='color_'+current_color
>       text_box.tag_config(tag_name, foreground=current_color)
>       for tag in text_box.tag_names(selected_text[0]):
>         text_box.tag_remove(tag,selected_text[0],selected_text[1] #We remove a new tag in case there is already an existing tag
>       text_box.tag_add(tag_name, selected_text[0],selected_text[1]) #We add the tag to the selected text with the new foreground color
> 
> option_menu=ctk.CTkOptionMenu(root, values=[x for x in colors], command=lambda color: change_color(color))
> option_menu.pack(pady=10)
> ```
> 
> I'm not sure if this is particularly what you want but this is how I used the tag methods when working with the textbox widget. Hope it helps!

This works!! Is there a way to make the text bold, italics and underline using this?

**Comment by Kurzuxus** [comment]
Yes, there is. This is how I made it work but one downside is it applies to all the text not just the text you highlight
```
Styles=["bold", "italic", "underline", "normal"] # Create a list containing styles
text_box=ctk.CTkTextbox(nest_fr2,width=400,height=300,wrap='word') Create a textbox widget
text_box.pack(pady=10)

def change_style(self,style):
        current_style=style_option.get() # We grab the current style from the option list
        text_box.configure(font=('Arial',12,current_style)) #Configure the text inside the textbox
#The font accepts the arguments in the following order: font(font_name, font_size, font_style) 

style_option=ctk.CTkOptionMenu(root, values=[x for x in Styles], command=lambda style: change_style(style))
style_option.pack(pady=10)
```

**Comment by Kurzuxus** [comment]
> > I have been working on a personal project and I did encounter an issue similar to this. I managed to do it this way and hopefully it will work for you:
> > ```python
> > text_box=ctk.CTkTextbox(nest_fr2,width=400,height=300,wrap='word') #Create a textbox widget
> > text_box.pack(pady=10)
> > 
> > colors=['red','blue','green']
> > 
> > def change_color(color):
> >     selected text=text_box.tag_ranges('sel') #We get the range of highlighted text in the textbox
> >     current_color=option_menu.get() #The current color selected in the option menu
> >     if selected_text: # add this condition so when the file first starts it doesn't give an error that a text is not highlighted 
> >       tag_name='color_'+current_color
> >       text_box.tag_config(tag_name, foreground=current_color)
> >       for tag in text_box.tag_names(selected_text[0]):
> >         text_box.tag_remove(tag,selected_text[0],selected_text[1] #We remove a new tag in case there is already an existing tag
> >       text_box.tag_add(tag_name, selected_text[0],selected_text[1]) #We add the tag to the selected text with the new foreground color
> > 
> > option_menu=ctk.CTkOptionMenu(root, values=[x for x in colors], command=lambda color: change_color(color))
> > option_menu.pack(pady=10)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I'm not sure if this is particularly what you want but this is how I used the tag methods when working with the textbox widget. Hope it helps!
> 
> This works!! Is there a way to make the text bold, italics and underline using this?

Yes, there is. This is how I made it work but one downside is it applies to all the text not just the text you highlight
```
Styles=["bold", "italic", "underline", "normal"] # Create a list containing styles
text_box=ctk.CTkTextbox(nest_fr2,width=400,height=300,wrap='word') Create a textbox widget
text_box.pack(pady=10)

def change_style(self,style):
        current_style=style_option.get() # We grab the current style from the option list
        text_box.configure(font=('Arial',12,current_style)) #Configure the text inside the textbox
#The font accepts the arguments in the following order: font(font_name, font_size, font_style) 

style_option=ctk.CTkOptionMenu(root, values=[x for x in Styles], command=lambda style: change_style(style))
style_option.pack(pady=10)
```

**Comment by astronaut954** [comment]
> I have been working on a personal project and I did encounter an issue similar to this. I managed to do it this way and hopefully it will work for you:
> 
> ```python
> text_box=ctk.CTkTextbox(nest_fr2,width=400,height=300,wrap='word') #Create a textbox widget
> text_box.pack(pady=10)
> 
> colors=['red','blue','green']
> 
> def change_color(color):
>     selected text=text_box.tag_ranges('sel') #We get the range of highlighted text in the textbox
>     current_color=option_menu.get() #The current color selected in the option menu
>     if selected_text: # add this condition so when the file first starts it doesn't give an error that a text is not highlighted 
>       tag_name='color_'+current_color
>       text_box.tag_config(tag_name, foreground=current_color)
>       for tag in text_box.tag_names(selected_text[0]):
>         text_box.tag_remove(tag,selected_text[0],selected_text[1] #We remove a new tag in case there is already an existing tag
>       text_box.tag_add(tag_name, selected_text[0],selected_text[1]) #We add the tag to the selected text with the new foreground color
> 
> option_menu=ctk.CTkOptionMenu(root, values=[x for x in colors], command=lambda color: change_color(color))
> option_menu.pack(pady=10)
> ```
> 
> I'm not sure if this is particularly what you want but this is how I used the tag methods when working with the textbox widget. Hope it helps!

I'm confused with your solution, is it only to manually highlight text?
